proto/internal/temporal/server/api/deployment/v1/message.proto

```diff
@@ -48,4 +50,9 @@ message DeploymentNameWorkflowArgs {
     string namespace_name = 1;
     string namespace_id = 2;
     string default_build_id = 3;
+}
+
+message RegisterWorkerInDeploymentArgs{
+    string task_queue_name = 1;
+    DeploymentWorkflowArgs.TaskQueueFamilyInfo.TaskQueueInfo task_queue_info = 2;
```

Comment:
@ShahabT - I thought about changing this to just the task-queue-type but decided against it mainly because of the following reason: In our deployment workflow, we want to store each registered task-queue with the poller time which shall come from matching. I haven't included in the code (right now) what FirstPollerTime will be, but I assume we can't develop this value in the deployment workflow.

I think we should keep this as is, but as always open to hearing your thoughts

Reply:
TaskQueueInfo is part of the internal state of the workflow and something that is built and updated over time via this RegisterWorker updates or other things that happen to the workflow.

I don't have strong opinion about the source of the timestamp. If you want to send it from matching, it's a valid choice I think. But that does not mean we should mix the state of wf with input of this update. The input would be:
```
message RegisterWorkerInDeploymentArgs{
    string task_queue_name = 1;
    TaskQueueType task_queue_type = 2;
    Timestamp poller_timestamp = 3;
}
```

Reply:
hmm, no hard objections and shall convert this. thanks for the discussion!